docs/source/mods/metromap.rst

```diff
@@ -0,0 +1,585 @@
+In order to use this functionality, the ``plotly`` package is needed.
+
+The plot function generates a plot that is opened in a browser. The lines can be
+turned off and on in this plot when clicking the respective name in the legend.
+The plot function can be called as follows::
+
+    from gurobi_optimods.metromap import plot_map
+    plot_map(graph_out, edge_directions, linepath_data)
+
+The following figure is an example of the above call, it shows the lines in the
+octilinear representation of the Berlin metro network.
+(Use `link <../_static/metro_berlin.html>`_ to open in full screen.)
+
+.. raw:: html
+
+    <iframe src="../_static/metro_berlin.html" width="100%" height="600"></iframe>
+
+
+Note that plotting the lines in the octilinear network is an art or a further
+optimization problem. If lines share an edge, it needs to be decided how to shift
+the lines so that both can be distinguished, which is above, below, left, or
+right, and which colors should be used.
+The plotting function that is provided in this OptiMod is not very sophisticated.
+Feel free to contribute with improvement ideas.
+
+
+Parameter: Planarity and Objective Function
+-------------------------------------------
+
+As already mentioned, it is possible to change the weights in the objective or
+omit the planarity restrictions. The planarity parameter was already considered
+in the solve example. The default value is true, i.e., if the parameter is not
+set when calling the ``metromap`` function, the planarity constraints are
+included via a callback function.
+
+Similar holds for the weights of the different parts in the objective. The
+default value for all weights is 1. If a different weighting is requested, this
+can be done as follows::
+
+  graph_out, edge_directions = metromap(
+      graph,
+      linepath_data,
+      penalty_edge_directions=2,
+      penalty_line_bends=0,
+      penalty_distance=1,
+  )
+
+
+Combination with Line Optimization OptiMod
+------------------------------------------
+After computing an optimal line plan with the Line Optimization OptiMode, an
+octilinear representation and the corresponding metromap can be computed with
+this OptiMod. Here is an example of how this could be done
+
+
+.. testcode:: combine
+  :skipif: size_limited_license()
+
+  # import all requirements
+  import networkx as nx
+  from gurobi_optimods import datasets
+  from gurobi_optimods.line_optimization import line_optimization
+  from gurobi_optimods.metromap import metromap
+  from gurobi_optimods.metromap import plot_map
+
+  # load data for line optimization and compute line plan
+  node_data, edge_data, line_data, linepath_data, demand_data = (
+      datasets.load_siouxfalls_network_data()
+  )
+  frequencies = [1, 3]
+  obj_cost, final_lines = line_optimization(
+      node_data,
+      edge_data,
+      line_data,
+      linepath_data,
+      demand_data,
+      frequencies,
+      True,
```

Comment:
Use a keyword arg here for clarity: `shortest_paths=True`

Reply:
In light of #150 - if `True` is the default and might change in future, I suggest just removing the argument in this example.